pkg/addons/embeddedclusteroperator/embeddedclusteroperator.go

```diff
@@ -263,43 +263,58 @@ func (e *EmbeddedClusterOperator) Outro(ctx context.Context, provider *defaults.
 		},
 		Spec: ecv1beta1.InstallationSpec{
 			ClusterID:                 metrics.ClusterID().String(),
-			MetricsBaseURL:            metrics.BaseURL(license),
+			MetricsBaseURL:            metrics.BaseURL(e.license),
 			AirGap:                    e.airgap,
 			Proxy:                     proxySpec,
 			Network:                   k0sConfigToNetworkSpec(k0sCfg),
 			Config:                    cfgspec,
 			RuntimeConfig:             e.runtimeConfig,
 			EndUserK0sConfigOverrides: euOverrides,
-			BinaryName:                defaults.BinaryName(),
+			BinaryName:                e.binaryName,
 			LicenseInfo: &ecv1beta1.LicenseInfo{
-				IsDisasterRecoverySupported: licenseDisasterRecoverySupported(license),
+				IsDisasterRecoverySupported: licenseDisasterRecoverySupported(e.license),
 			},
 		},
 	}
 	if err := cli.Create(ctx, &installation); err != nil {
 		return fmt.Errorf("unable to create installation: %w", err)
 	}
+
+	gotInstallation, err := waitForInstallationToExist(ctx, cli, installation.Name)
```

Comment:
i cant understand at all why you have this right after creating it. you will need a comment here to explain it.

Reply:
Because otherwise it's a race condition with the apiserver

Reply:
comment added